packages/token/src/index.ts

```diff
@@ -134,12 +159,40 @@ export class Token {
         return formatUnits(amount, this.decimals);
     }
 
+    /**
+     * Get a token from a ticker on a specific chain.
+     *
+     * @param ticker The ticker of the token to get.
+     * @param chain The chain to get the token from.
+     */
+    static fromTickerOnChain(ticker: string, chain: ChainId): Token {
+        if (!Token.tokenMappings[chain]) {
+            throw new Error(
+                `Token mapping for chain ${chain} not initialized. Call fetchRemapFromRepo first.`,
+            );
+        }
+
+        const tokenData = Token.tokenMappings[chain].tokens.find(
+            (token) => token.ticker === ticker,
+        );
+        if (!tokenData) {
+            return DEFAULT_TOKEN;
+        }
+
+        return new Token(tokenData);
+    }
+
+    /** @deprecated Use `Token.fromTickerOnChain` instead */
```

Comment:
Don't think this needs to be deprecated, it can be used in single chain contexts or when the address doesn't matter (maybe you just want decimals).